### PR TITLE
fix(patch): [sc-10502] Encode remote target names as integer.

### PR DESCRIPTION
**Pull request by ser-0xff**
## Description

Encode remote target names as integer to reduce e amount of transmitted data.

## How Has This Been Tested?

Unit tests.

## Minimal checklist:

- [X] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by ser-0xff** [comment]
So, with full function name:
image

**Comment by ser-0xff** [comment]
And encoding function name as integer gives us:
image

**Comment by hassila** [comment]
And with compression now with the optimization?

**Comment by ser-0xff** [comment]
Yes, I had to provide values with and without compression for both cases.
Here is it function names:
image
Without function names:
image

**Comment by hassila** [comment]
Thanks, still 3x for that simple message, lets see what we get on our more complex real-world payloads with more mixed data (and also lets try it with dictionary training as discussed).

File: Sources/DistributedSystem/DistributedSystem.swift [hunk 1]
```diff
@@ -1023,17 +1024,27 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
         let payloadSize =
             MemoryLayout<SessionMessage.RawValue>.size
                 + ULEB128.size(actor.id.instanceID)
-                + InvocationEnvelope.wireSize(callID, target, invocation.genericSubstitutions, invocation.arguments)
+                + InvocationEnvelope.wireSize(callID, invocation.genericSubstitutions, invocation.arguments, target)
         // Even if we carefully calculated the capacity of the desired buffer and know it to the nearest byte,
         // swift-nio still allocate a buffer with a storage capacity rounded up to nearest power of 2...
```
[the remark this answers is not in the document]
Comment (hassila) [reply Sources/DistributedSystem/DistributedSystem.swift:1029]: Will you file an issue please?

**Comment by ktoso** [comment]
Looking good, I don't know all your assumptions in this system implementation but I like how simple the approach is. So I'm assuming you don't really consider message drops; like if the first message with string/id is dropped then continuing to send next one with just id may not be right; but I assume you don't handle such reconnects to the same peer then? Or I'm missing some other context.

Either way, yeap, that's the general idea! It's great to see it applied, depending on necessity things may become more defensive (in Akka this was a bunch more complext), but it's quite a huge save on payload size, so it's great to see folks already apply the technique 👍

**Comment by ser-0xff** [comment]
Thank you for the review, @ktoso.
Why do you think the message can be dropped?

**Comment by ktoso** [comment]
I don't know your transport mechanism in this lib, in my world I'd always worry about message drops and reconnects to the same peer. :) Perhaps it doesn't apply here

**Comment by ser-0xff** [comment]
The idea was to have a dedicated mapping per NIO channel, in that case we can avoid contention on that mapping container.
So mapping is tied to particular NIO Channel and we build it from the scratch for each new connection. We do not expect too many connections in our Distributed System application, so I think it is justified to pay with memory for speed.

**Comment by ktoso** [comment]
Yeah, neat ideat :) Tying the tables to the channels like this makes it much simpler, I quite like it.

**Comment by hassila** [comment]
Do we want a unit test for this?

**Review by hassila — APPROVED** [review]

**Comment by ser-0xff** [comment]
What unit test would check from your point of view?

**Comment by ser-0xff** [comment]
I thought you have some ideas...
IMHO all unit tests we have now also covers that functionality because it works always.

**Pull request merged**